llmfoundry/callbacks/async_eval_callback.py

```diff
@@ -205,79 +229,186 @@ def __init__(
-                                          self.training_config['save_interval'])
-        self.check_interval = create_interval_scheduler(
-            interval,
-            # There is a custom close to ensure that the final checkpoint
-            # (which is the most important) is evaled after it is written
-            include_end_of_training=False,
-        )
-        self.compute = compute
-        self.last_checkpoint: Optional[str] = None
 
-        # Run these during init to fail fast in any of the error cases
         self.current_run = self._get_current_run()
         get_eval_parameters(
             parameters=training_config,
             checkpoint='test',
             training_run_name=self.current_run.name,
         )
-        log.info(
-            f'Initialized AsyncEval callback. Will generate runs at interval {interval}'
+
+        # Validate the interval (how often to launch eval runs)
+        self.interval = validate_interval(interval,
+                                          self.training_config['save_interval'])
+
+        # Validate and compute the check interval (how often to check for new checkpoints)
+        if check_interval is None:
+            unit = self.interval.value // 5
+            if unit == 0:
+                unit = 1
+            check_interval = Time(unit, self.interval.unit)
+            log.info(
+                f'No check interval provided, defaulting to {check_interval}')
+
+        self.check_interval = validate_check_interval(check_interval,
+                                                      self.interval)
+
+        # Keep track of checkpoints by interval that have already been evaled
+        # Format: {interval: (checkpoint, run_name)}
+        self.checkpoints_evaled: Dict[Time, Tuple[str, str]] = {}
+
+        # Scheduling is based on the check interval, while _get_checkpoints_and_launch_runs
+        # will only launch runs at the interval
+        self.is_at_check_interval = create_interval_scheduler(
+            self.check_interval,
+            # There is a custom close to ensure that the final checkpoint
+            # (which is the most important) is evaled after it is written
+            include_end_of_training=False,
         )
 
+        log.info('Initialized AsyncEval callback. Will generate runs at ' +
+                 f'interval {interval}, checking at {check_interval}')
+
+    def _get_checkpoints_and_launch_runs(self, state: State):
+        """Get the latest checkpoint from the training run.
+
+        Args:
+            state: The current state of the training run
+
+        Returns:
+            Returns checkpoints that have not been evaled
+        """
+        checkpointer = None
+        for callback in state.callbacks:
+            if isinstance(callback, CheckpointSaver):
+                checkpointer = callback
+                break
+
+        if not checkpointer:
+            log.warning('No checkpoint saver callback found. Skipping eval')
```

Comment:
this should be `warnings.warn` I think, because it is a warning the user could take corrective action to fix